### DISCUSSION TITLE
Solar radiation, Albedo,... results print out

### DISCUSSION THREAD

**Discussion by Bemtgenj** [groops-devs/groops#220] [Q&A]
Hello 

Is there an option in groops to print out results for final values of for example Solar Radiation, albedo,TRP ... internal parameters (similar to results for final position, velocity, acceleration over time)? I have looked through the documentation and found among other places some information in the parameter name section on page 29, the  NormalsCreate and the ParameterNamesCreate. but i am not really sure if sure how the name needs to the spelled out (see 2.type below) for i.e: SolarRadiationPressure.x

The parameter is given by explicitly by four parts:
1. object: Object this parameter refers to, e.g. graceA, G023, earth, . . .
2. type: Type of this parameter, e.g. accBias, position.x, . . .
3. temporal: Temporal representation of this parameter, e.g. trend, polynomial.degree1, . . .
4. interval: Interval/epoch this parameter represents, e.g. 2017-01-01_00-00-00_2017-01-02_00-00-00,
2018-01-01_00-00-00.

and then after GNSSProcessing
run ParamaterNamesCreate and NormalsCreate...?

I will also continue trying with the GRACEFO base example.
Thanks for the Help

Cheers
Jean

**Comment by tmayerguerr** [comment]
Hi Jean

If you put after the the last estimation a [processingStep:writeAprioriSolution](https://groops-devs.github.io/groops/html/gnssProcessingStepType.html#writeAprioriSolution) in [GnssProcessing](https://groops-devs.github.io/groops/html/GnssProcessing.html) , you can write with `outputfileAprioriSolution` the full solution vector (used as apriori for the possible next iteration). With `outputfileParameterNames` file the order and meaning of the parameters are explained.

Cheers
Torsten

**Comment by Bemtgenj** [comment]
Hallo Torsten

Thank you for the feedback and the suggestion. I will try that, if i have more questions i will let you know.

Cheers
Jean

**Reply by Bemtgenj** [reply]
Hallo Torsten,
That worked fine - I do get now the values for D0, DC2, DS2,  ,… . However, to do the final calculation for the SRP acceleration I will also need Δu = usun – usat. Is that value as well in the parameter list, not by default but has to be defined i.e. the remaining parameters part of the writeAprioriSolution processing step?
Can i use these as parameter names?
 **Vector3d posSat** [= rotSat.inverseRotate(position);]
 **Vector3d posSun** [ = rotSat.inverseRotate(ephemerides->position(time, Ephemerides::SUN));]

with this one being equivalent to deltau then?
 Vector3d    **directionSunSat**  { posSat-posSun;
    const Double distanceSunSat  = directionSunSat.normalize();]

or can i use this variable and it is the acceleraton vector for SRP already in a satellite sun reference frame?

**Vector3d F** [= factorSun * force(satellite, directionSunSat, ny*solarflux/LIGHT_VELOCITY*std::pow(AU/distanceSunSat, 2), 0., absorbedPressure);]

i got the Remaining parameters part working with for example solarRadiationPressure.ECOM.D0 or xxx.DC2,.. but not yet for the aforementioned parameters. Is there maybe a parameter list somewhere other than fileParameterName.h which i think is the general parameter list?
![All_script_1](https://github.com/groops-devs/groops/assets/100558965/739659e7-ab6c-4425-9f5c-933bc9cad653)




![image](https://github.com/groops-devs/groops/assets/100558965/68598370-091e-406b-b336-8ce4b0600d65)


 i also noticed that i get somewhat different  values for the same parameter if i run the script with for example the 2 parameters above and no parameters defined (all of them then). Shouldn't they be the same as there is no change in script per say.

![Two_Name_1](https://github.com/groops-devs/groops/assets/100558965/e8bdb22e-1a48-460d-8a64-92ef20fc39a4)

![Two_Values_1](https://github.com/groops-devs/groops/assets/100558965/40d0e858-fa0b-4f14-8dc4-ad7d0cf9bab7)
![All_Name_1](https://github.com/groops-devs/groops/assets/100558965/b4db8ea5-2593-4d2d-9af4-aed64cfd6d41)

![All_Values_1](https://github.com/groops-devs/groops/assets/100558965/7c5659ba-73fb-4309-9cde-b7c2656a929f)

Txs for the help
Cheers
Jean

**Comment by tmayerguerr** [comment]
Hi Jean,

1. With [ParameterSelection2IndexVector](https://groops-devs.github.io/groops/html/ParameterSelection2IndexVector.html) and [MatrixCalculate](https://groops-devs.github.io/groops/html/MatrixCalculate.html)->[reorder](https://groops-devs.github.io/groops/html/matrixGeneratorType.html#reorder) you can select your parameters afterwards
2. The estimated SRP parameters can be converted to accelerations with [InstrumentAccelerometerApplyEstimatedParameters](https://groops-devs.github.io/groops/html/InstrumentAccelerometerApplyEstimatedParameters.html)

> i also noticed that i get somewhat different values for the same parameter if i run the script with for example the 2 parameters above and no parameters defined (all of them then). Shouldn't they be the same as there is no change in script per say.

The results should be the same. You just looked in the wrong line. In your first example the parameter names in the text line 5 and 6 agree with the values in matrix/vector in lines 3 and 4 (Different size of header).

Best regards
Torsten

**Comment by Bemtgenj** [comment]
Hi Torsten

txs for the quick feedback, i will try your suggestion.

Comparing the wrong lines :-( .......did not think of the headers,  that's what i thought,.. it did not make sense that they were not the same 

Cheers
Jean

**Reply by Bemtgenj** [reply]
Hello Torsten,

Thank you so much for your help so far

i have a couple more questions wrt to this topic. I think i have done correctly what you have suggested above, but i am not sure i did the last step correctly

![Screenshot from 2023-11-07 09-21-41](https://github.com/groops-devs/groops/assets/100558965/b96b85eb-5fa3-4a27-9e0e-4035a5c32c2d)
![Screenshot from 2023-11-07 09-23-05](https://github.com/groops-devs/groops/assets/100558965/dfc26060-03bd-446b-ad9b-e80b2f3f7ad2)

If i understand the InstrumentAccelerometerApplyEstimatedParamters script correctly then i need to
- add the satellite model
- add the orbits file
- add gnssSolarRadiation
- add x0_reordered file for the parameters (solarRadiation parameters)

The output will be in CRF frame as the GNSSProcessing.

I want to make sure this is correct as i have not yet found any comparable output data that would verify my output results results are correct.

Is the StarCamera file needed as well? Is the earth rotation file needed to change reference frames (TRF/CRF)? 

Thank you very much  for your feedback and help.

Cheers
Jean

**Comment by tmayerguerr** [comment]
Hi Jean,

I have only just seen that you are using the gnssSolarRadiationPressure model for GRACE. However, this is not suitable as it is based on a boxwing model with an orientation of the satellite towards the sun. This also requires the attitude of the satellite (inputfileStarCamera). The output of InstrumentAccelerometerApplyEstimatedParamters is usually in the Satellite Reference Frame (SRF).

By the way, I have improved the documentation so that the parameter names are also described.

regards
Torsten

**Reply by Bemtgenj** [reply]
Hi Torsten,

txs for the feedback - that is a good point wrt to box-wing model, i mainly thought of using GRACE as most information is available for that satellite as a "practice" satellite before moving onto LEO satellites.

The other point that came up now looking at it again in more details  is that i also noticed that using acceleration type (in parametrizationAccelaration) gnssSolarRadiation is for GNSS satellites only and not for any LEO satellite missions? So i think i was using the wrong parameter as well ... and i should use for a LeoSat mission the following setup

![Solarad](https://github.com/groops-devs/groops/assets/100558965/298c3e32-cd21-48a2-8260-4c27c28c2457)

to use the same solar radiation parameters as in the variational equations script.
i would need to find a LEO satellite that is boxwing type and has the satellite model to do some "practice calculations".

The plan was to run solar radiation on GNSS satellites to get some results to check the scripts and as well on a "known" LEO satellite mission.

Using the GRACE mission as i was doing before brings up another question then - the non-conservative forces on the TUG web include all parameters together? Is then possible to run the script to get only the solar radiation portion?

Txs for the continuous help and advises and also updating the documentation :-)



Cheers
Jean

**Reply by Bemtgenj** [reply]
23 Nov 23 update

i redid the solar radiation on a CubeSat with modelScale:solarRadiation and got in the parNam file the following output:
sat.modelScale:: with no other information. is there no parameter for this type of solar radiation?

For GNSS i get all the gnssSolarRadiation parameters for all  the for example the GPS satellites, i just now need to figure out how to loop through them to get all of into the indexvector file.

Thank you for your continuous help and advice
Cheers
Jean

**Reply by tmayerguerr** [reply]
modelScale means it evaluates/compute the model values (in this case solarRadiation) and apply the model with an estimated factor to the data. Therefore the unknown parameter is one single unitless factor.
regards
Torsten

**Comment by Bemtgenj** [comment]
Hello Torsten

Txs very much for the feedback.

So if i understand correctly your feedback the modelScale does compute the (in our example the solarRadiationPressure for a given satellite) and in the nominal case the factor is 1 so the unknown parameter would be 1 in this case? 

Then i am not clear on how to get the non-conservative forces and for example more specifically SRP. My idea  was to use modelScale multiple times once with SRP, albedo, atmospheric Drag,... and so on to get the different force parameters , knowing in which order i had defined them in the script (as they were not specifically called out as a parameter).  How are the conservative forces files on the TUG website then created as those different parameters are needed for those files?

i will dig some more into this as well and let you know if i get any further, ( i have managed to get the acceleration on the GNSS satellites calculated, checking now if they are correct - see example below)
![Screenshot from 2023-11-29 07-50-01](https://github.com/groops-devs/groops/assets/100558965/ca57fb32-5e5d-468e-8eec-1d72f774dcfa)

I would like to thank you again for the big help you have providing by helping out with my questions

Cheers
Jean
